ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/manager/checkpoint/CheckpointOffsetManager.scala

```diff
@@ -31,38 +32,30 @@ private[manager] object CheckpointOffsetManagerProps {
   val STARTING_OFFSETS_EARLIEST = "earliest"
 }
 
-private[manager] class CheckpointOffsetManager(val topic: String, val checkpointBaseLocation: String,
-                                               val startingOffsets: Option[String]) extends OffsetManager(topic) {
+private[manager] class CheckpointOffsetManager(val checkpointLocation: String,
+                                               val startingOffsets: Option[String]) extends OffsetManager("") {
 
   import CheckpointOffsetManagerProps._
 
-  if (StringUtils.isBlank(topic)) {
-    throw new IllegalArgumentException(s"Invalid topic: '$topic'")
-  }
-
-  if (StringUtils.isBlank(checkpointBaseLocation)) {
-    throw new IllegalArgumentException(s"Invalid checkpoint base location: '$checkpointBaseLocation'")
+  if (StringUtils.isBlank(checkpointLocation)) {
+    throw new IllegalArgumentException(s"Invalid checkpoint location: '$checkpointLocation'")
   }
 
   private val logger = LogManager.getLogger
 
-  private val checkpointLocation = resolveCheckpointLocation(topic)
-
   override def configureOffsets(streamReader: DataStreamReader, configuration: org.apache.hadoop.conf.Configuration): DataStreamReader = {
```

Comment:
Maybe as part of another PR, changing from `configureOffsets` to `configure` would be semantically more correct and also more generic.

Reply:
Absolutely. Will do it in #87